spec/operators/startWith-spec.js

```diff
@@ -1,20 +1,93 @@
-/* globals describe, it, expect */
+/* globals describe, it, expect, expectObservable, hot, rxTestScheduler */
 var Rx = require('../../dist/cjs/Rx');
 var Observable = Rx.Observable;
 
 describe('Observable.prototype.startWith()', function () {
-  it('should start an observable with given value', function (done) {
-    var source = 'source'
-    var init = 'init';
-    var expected = [init, source];
-
-    var i = 0;
-    Observable.of(source)
-      .startWith(init)
-      .subscribe(function (x) {
-        expect(x).toBe(expected[i++]);
-      }, null, function () {
-        done();
-      });
-  });
-});
+  var defaultStartValue = 'x';
+  
+  it('should start an observable with given value', function() {
+    var e1 =   hot('--a--|');
+    var expected = 'x-a--|';
+    
+    expectObservable(e1.startWith(defaultStartValue)).toBe(expected);
+  });
+  
+  it('should start with given value and does not completes if source does not completes', function() {
+    var e1 =   hot('----a-');
+    var expected = 'x---a-';
+    
+    expectObservable(e1.startWith(defaultStartValue)).toBe(expected);
+  });
+  
+  it('should start with given value and does not completes if source never emits', function() {
+    var e1 = Observable.never();
+    var expected = 'x-';
+    
+    expectObservable(e1.startWith(defaultStartValue)).toBe(expected);
+  });
+  
+  it('should start with given value and completes if source does not emits', function() {
+    var e1 =   hot('---|');
+    var expected = 'x--|';
+    
+    expectObservable(e1.startWith(defaultStartValue)).toBe(expected);
+  });
+  
+  it('should start with given value and complete immediately if source is empty', function() {
+    var e1 = Observable.empty();
+    var expected = '(x|)';
+    
+    expectObservable(e1.startWith(defaultStartValue)).toBe(expected);
+  });
+  
+  it('should start with given value and source both if source emits single value', function() {
+    var e1 = Observable.of('a');
+    var expected = '(xa|)';
+    
+    expectObservable(e1.startWith(defaultStartValue)).toBe(expected);
+  });
+  
+  it('should start with given values when given value is more than one', function() {
+    var e1 =   hot('-----a--|');
+    var expected = '(yz)-a--|';
```

Comment:
And there is no other way, because otherwise if the parenthesis group would push the diagram to the right it would mean the parenthesis group would **not** be synchronous at that single frame.